specs/LaunchingRegisteredProtocols.md

```diff
@@ -0,0 +1,318 @@
+# Background
+
+We are exposing an event that will be raised when an attempt to launch a protocol that is registered with the OS (external protocol) is made.

```

Comment:
General terminology question. How do I think about "Protocol" vs "URI Scheme" in the CoreWebView2 code base?  I would be inclined to describe the entire process of locally launching (or blocking) "calculator:/" through webview in terms of "URI Scheme". But my experience of scoping "Protocol" to cross-device communication may not be correct here.

Reply:
Great question. We're not consistent and the terminology has kind of mixed everywhere.

I would say the most accurate way to define it is that for something like `https://example.com/`, the URI scheme is 'https' and the associated protocol is HTTP over TLS. That is, the URI scheme is from the initial characters of the URI, and the URI scheme tells you what (if any) network protocols to use in order to resolve the URI. I mostly think of networking protocol when I read protocol so for cases like 'search:' or any other URI scheme registered with the OS by an application I don't think any 'protocol' is involved (but how we define protocol here is debatable.)

Windows calls these 'registered protocols' in UWP: [Modern Apps Protocol Registration and Activation in UWP Apps](https://learn.microsoft.com/en-us/archive/msdn-magazine/2017/september/modern-apps-protocol-registration-and-activation-in-uwp-apps), `Windows.ApplicationModel.Activation.ActivationKind.Protocol`, the syntax in the mainfest is `windows.protocol`. And before UWP from the Windows shell we had 'application protocols' and from urlmon we had 'asynchronous pluggable protocols'. See [app protocols](https://learn.microsoft.com/en-us/previous-versions/windows/internet-explorer/ie-developer/platform-apis/aa767914(v=vs.85)) or ['search:'](https://learn.microsoft.com/en-us/windows/win32/shell/search-protocol). But plenty of those documents and documents nearby also refer to the URI scheme when trying to actually define or explain things.

This is also confused by the early days of the browser. The DOM API `window.location.protocol` gets you the current document's URI's scheme name. More recently they created a URL type that you can use to parse URIs generically (not just the current documents URI) and for compat they had to retain the weird early property names. So the spec for  `URL.protocol` says: 

> The protocol getter steps are to return [this](https://webidl.spec.whatwg.org/#this)’s [URL](https://url.spec.whatwg.org/#concept-url-url)’s [scheme](https://url.spec.whatwg.org/#concept-url-scheme), followed by U+003A (:).

Oops. (Along the same lines they call the URI's fragment `hash`, and the URI's query `search`...)

Anyway, its confused enough that I would be fine calling this whichever (protocol or URI scheme) we think will be the easiest for end devs to understand (regardless of how I clearly feel about it 😅). 

We've gone with protocol here because it better matches what Windows calls these. Although it will be good to see what Linux and MacOS call them since WebView2 will also ship there.

Reply:
Does that mean changing this event from `LaunchingExternalProtocol` to `LaunchingUriSchemeIdentified`?

Reply:
@maurawinstanley does this event apply to all URI schemes that WebView2 doesn't know about, or only URI schemes that are app URI schemes registered with the OS that WebView2 doesn't otherwise know about?

Reply:
This only applies to URI schemes that are app URI schemes registered with the OS that WebView2 doesn't otherwise know about

Reply:
Discussion: we might add a unknown URI scheme event in the future. Let's name it with 'UriScheme' to help match with that in the future:
 * `LaunchingExternalUriScheme`

Reply:
@david-risney should this be LaunchingExternalURIScheme with URI in all caps?

Reply:
Capitalization rules are that 2-letter acronyms use all-caps, but 3-or-more letter acronyms capitalize only the first letter. So Uri.

Reply:
@oldnewthing does this also apply to URI outside of the event name, ie if URI is just used in a sentence?

Reply:
It should be 'Uri' in API names and 'URI' in our documentation.